core/indexer/common.go

```diff
@@ -246,16 +246,17 @@ func serializeTransactions(
 	selfShardID uint32,
 	_ func(hashes []string, index string) (map[string]bool, error),
 	mbsHashInDB map[string]bool,
-) []bytes.Buffer {
+) ([]bytes.Buffer, error) {
 	var err error
 
 	var buff bytes.Buffer
 	buffSlice := make([]bytes.Buffer, 0)
 	for _, tx := range transactions {
 		isMBOfTxInDB := mbsHashInDB[tx.MBHash]
-		meta, serializedData := prepareSerializedDataForATransaction(tx, selfShardID, isMBOfTxInDB)
-		if len(meta) == 0 {
-			continue
+		meta, serializedData, errPrepareTx := prepareSerializedDataForATransaction(tx, selfShardID, isMBOfTxInDB)
+		if errPrepareTx != nil {
+			log.Warn("error preparing transaction for indexing", "tx hash", tx.Hash, "error", err)
```

Comment:
We need a task (that will be done in a next PR) to further refactor and remove all these warns as for this error we will get 3 log.Warn prints.

Reply:
ok.. good catch. added jira ID EN-7977